src/pre_process/include/1dHardcodedIC.fpp

```diff
@@ -8,6 +36,42 @@
     select case (patch_icpp(patch_id)%hcid)
     case (100)
         ! Put your variable assignments here
+        if (.not. files_loaded) then
+            ! Print status message
+            print *, "Loading all data files..."
+            do f = 1, nFiles
+                ! Open the file for reading
+                open (newunit=unit, file=trim(fileNames(f)), status='old', action='read', iostat=ios)
+                if (ios /= 0) then
+                    print *, "Error opening file: ", trim(fileNames(f))
+                    cycle  ! Skip this file on error
+                end if
+                ! Read all rows at once into memory
+                do iter = 1, nRows
+                    read (unit, *, iostat=ios) x_coords(iter), stored_values(iter, f)
+                    if (ios /= 0) then
+                        print *, "Error reading file ", trim(fileNames(f)), " at row ", iter
+                        exit  ! Exit loop on error
```

Comment:
don't use print like this, it will print 1000 lines of same line if you use 1000 MPI ranks. 

don't use exit or stop. use s_mpi_abort('Reason for exit')